client/src/app/submission/submission.component.ts

```diff
@@ -96,7 +129,7 @@ export class SubmissionComponent implements OnInit, OnDestroy {
       this.submissionEnvelopeId = SubmissionComponent.getSubmissionId(submissionEnvelope);
       this.isValid = this.checkIfValid(submissionEnvelope);
       this.submissionState = submissionEnvelope['submissionState'];
-      this.isSubmitted = this.isStateSubmitted(submissionEnvelope.submissionState);
+      this.isSubmitted = this.isStateSubmitted(SubmissionState[submissionEnvelope.submissionState]);
```

Comment:
This function is very long. I would break it down to a few logical parts so it is more readable.

Reply:
Done.